eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -244,7 +252,7 @@ class Setup(val datadir: File,
 
       finalPubkey = new AtomicReference[PublicKey](null)
       pubkeyRefreshDelay = FiniteDuration(config.getDuration("bitcoind.final-pubkey-refresh-delay").getSeconds, TimeUnit.SECONDS)
-      bitcoinClient = new BitcoinCoreClient(bitcoin) with OnchainPubkeyCache {
+      bitcoinClient = new BitcoinCoreClient(bitcoin, if (bitcoin.wallet == onChainKeyManager_opt.map(_.wallet)) onChainKeyManager_opt else None) with OnchainPubkeyCache {
```

Comment:
We need this to load an onchain key manager that cab be used to generate descriptors through the API but is not active because wallet names in `eclair.conf` and `eclair-signer.conf` are different.

Reply:
This is not at all obvious and deserves a comment.

I think this may also clash with #2737 but we can figure this out when we merge that PR.